clang/lib/Sema/SemaBounds.cpp

```diff
@@ -3684,21 +3684,33 @@ namespace {
       Expr *SubExpr = E->getSubExpr()->IgnoreParens();
 
       if (isa<CompoundLiteralExpr>(SubExpr)) {
+        // The lvalue bounds of a struct-typed compound literal expression e
+        // are bounds(&value(temp(e), &value(temp(e)) + 1).
+        if (E->getType()->isStructureType()) {
+          Expr *TempUse = CreateTemporaryUse(E);
+          Expr *Addr = CreateAddressOfOperator(TempUse);
+          const llvm::APInt One(Context.getTargetInfo().getPointerWidth(0), 1);
+          IntegerLiteral *Size = CreateIntegerLiteral(One);
+          CountBoundsExpr *CBE = new (Context)
+              CountBoundsExpr(BoundsExpr::Kind::ElementCount, Size,
+                              SourceLocation(), SourceLocation());
+          return ExpandToRange(Addr, CBE);
```

Comment:
There's a prebuilt Count(1) expression available in ASTContext.  See ASTContext::getPrebuiltCountOne().   This avoids an extra allocation in ASTContext.  The code constructs a count(1) expression, but then calls ExpandToRange, which produces a range expression.